python/ray/data/_internal/execution/execution_callback.py

```diff
@@ -60,12 +60,16 @@ def get_execution_callbacks(context: DataContext) -> List[ExecutionCallback]:
         _initialize_env_callbacks(context)
         context.set_config(ENV_CALLBACKS_INITIALIZED_KEY, True)
 
+    from ray.data._internal.execution.callbacks.epoch_idx_update_callback import (
+        EpochIdxUpdateCallback,
+    )
     from ray.data._internal.execution.callbacks.insert_issue_detectors import (
         IssueDetectionExecutionCallback,
     )
 
     return context.get_config(
-        EXECUTION_CALLBACKS_CONFIG_KEY, [IssueDetectionExecutionCallback()]
+        EXECUTION_CALLBACKS_CONFIG_KEY,
+        [IssueDetectionExecutionCallback(), EpochIdxUpdateCallback()],
```

Comment:
Users can clear this list manually (and Ray Data emits some messages telling users how to do that if they want to disable the issue detectors), which will then break things. Should we always include this callback regardless of what the user set?

Reply:
Oh wait nvm we don't ever tell the user to clear this list -- I was thinking of this: https://github.com/ray-project/ray/blob/5328b3250f8a15dc35c391869c24f0db0f821a32/python/ray/data/_internal/issue_detection/issue_detector_manager.py#L97

But it's still possible for users to do this:

```python
ray.data.DataContext.get_current().set_config("execution_callbacks", [])
```